SALTO-1715: moved deployment annotations to core (#2556)
salto-io/salto: docs/syntax.md

}
```

#### _creatable
This is a hidden boolean annotation (will not be seen in NaCl) that is used to set whether creating new instances with the type or fields is supported.

Type: `boolean`
Default: `true`
Applicable to: Types, Fields

Example:

For the type
```HCL
type salto.example {
  number nonCreatableField {
    _creatable = false
  }
  number creatableField {
    _creatable = true
  }
  _creatable = true
}
```

The following new instance will be deployed without any error:
```HCL
salto.example valid {
  creatableField = 1
}
```

For the following new instance, an error will be thrown from a change validator on deploy:
```HCL
salto.example invalid {
  creatableField = 1
  nonCreatableField = 1
}

> ```

Author: Fixed the doc.
Note that currently if all the type does not support the operation we return an error and if only fields does not support the operation we return warning

}
```

#### _creatable

> This is a hidden boolean annotation (will not be seen in NaCl) that is used to set whether creating new instances with the type or fields is supported.

Reviewer: ```suggestion
This is a hidden boolean annotation (will not be seen in NaCl) that is used to set whether creating instances of a type or certain value in it is supported.
```

Type: `boolean`
Default: `true`
Applicable to: Types, Fields

Example:

For the type
```HCL
type salto.example {
  number nonCreatableField {
    _creatable = false
  }
  number creatableField {
    _creatable = true
  }
  _creatable = true
}
```

The following new instance will be deployed without any error:
```HCL
salto.example valid {
  creatableField = 1
}
```

> For the following new instance, an error will be thrown on deploy:

Reviewer: ```suggestion
For the following new instance, a warning will be shown in the deploy preview:
```

  creatableField = 1
  nonCreatableField = 1
}
```

#### _updatable
This is a hidden boolean annotation (will not be seen in NaCl) that is used to set whether a modification of an instance or a certain value in an instance is supported.

Type: `boolean`
Default: `true`
Applicable to: Types, Fields

Example:
For the types
```HCL
type salto.updatable {
  number nonUpdatableField {
    _updatable = false
  }
  number updatableField {
    _updatable = true
  }
  _updatable = true
}

type salto.notUpdatable {
  number someField {
  }
  _updatable = false
}
```

The following instance changes will be deployed without any error:
```HCL
// before
salto.updatable valid {
  updatableField = 1
}

// after
salto.updatable valid {
  updatableField = 2
}
```

```HCL
// before
salto.updatable valid {
  nonUpdatableField = 1
  updatableField = 1
}

// after
salto.updatable valid {
  nonUpdatableField = 1
  updatableField = 2
}
```

For the following instance change, an warning will be shown to the user before deploying:
```HCL
// before
salto.updatable invalid {
  nonUpdatableField = 1
  updatableField = 1
}

// after
salto.updatable invalid {
  nonUpdatableField = 2
  updatableField = 2
}
```

> For the following instance change, an error will be thrown to the user on deploy:

Reviewer: ```suggestion
For the following instance change, an error will be shown to the user on deploy and the change will not be deployed:
```

```HCL
// before
salto.notUpdatable invalid {
  someField = 1
}

// after
salto.notUpdatable invalid {
  someField = 2
}
```

#### _deletable
This is a hidden boolean annotation (will not be seen in NaCl) that is used to set whether a deletion of an instance is supported.

Type: `boolean`
Default: `true`
Applicable to: Types

Example:
For the types
```HCL
type salto.deletable {
  number someField {
  }
  _deletable = false
}

type salto.notDeletable {
  number someField {
  }
  _deletable = true
}
```

The deletion of the following instance will be deployed without any error:
```HCL
salto.deletable instance {
  someField = 2
}
```

> For the deletion of the following instance, an error will be thrown on deploy:

Reviewer: ```suggestion
For the deletion of the following instance, an error will be shown to the user and the change will not be deployed:
```

}
```

#### _creatable
This is a hidden boolean annotation (will not be seen in NaCl) that is used to set whether creating new instances with the type or fields is supported.

Type: `boolean`
Default: `true`
Applicable to: Types, Fields

Example:
```HCL
type salto.example {
  number nonCreatableField {
    _creatable = false
  }
  number creatableField {
    _creatable = true
  }
  _creatable = true
}

> ```

Reviewer: I think this is not a sufficient explanation of the behavior this annotation implies.
specifically it is not clear what this does for fields
